frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts

```diff
@@ -127,6 +172,174 @@ export class ExperimentDesignStepperService {
     return aliasTableData;
   }
 
+  createNewFactorialConditionTableData(designData: ExperimentFactorialDesignData): FactorialConditionTableRowData[] {
+    const tableData: FactorialConditionTableRowData[] = [];
+
+    // currently this table will only support 2 factors due to design constraints
+    // this will need revisited to support more factors in this table
+
+    const factorOne = designData.factors[0];
+    const factorTwo = designData.factors[1];
+
+    factorOne.levels.forEach((factorOneLevel) => {
+      factorTwo.levels.forEach((factorTwoLevel) => {
+        const tableRow: FactorialConditionTableRowData = {
+          id: uuidv4(), // TODO: maybe not the right place?
+          levels: [
+            {
+              id: factorOneLevel.id,
+              name: factorOneLevel.level,
+            },
+            {
+              id: factorTwoLevel.id,
+              name: factorTwoLevel.level,
+            },
+          ],
+          alias: this.createFactorialAliasString(
+            factorOne.factor,
+            factorOneLevel.level,
+            factorTwo.factor,
+            factorTwoLevel.level
+          ),
+          weight: '0.0',
+          include: true,
+        };
+
+        tableData.push(tableRow);
+      });
+    });
+
+    return tableData;
+  }
+
+  mergeExistingConditionsTableData(experimentInfo: ExperimentVM): FactorialConditionTableRowData[] {
+    const existingConditions = experimentInfo.conditions;
+    const existingConditionAliases = experimentInfo.conditionAliases;
+    const existingPartitions = experimentInfo.partitions;
+
+    const levelOrder = {};
+    existingPartitions.forEach((partition) => {
```

Comment:
I'm not sure this sort strategy will always work? Levels will always "tie" with the other levels in a factor, right? so what guarantees the order of levels within a factor?

We're sending the index as `order` to the backend for `factors`, "`partitions`", and `conditions`, so can we just add an `order` property in for `levelCombinationsElements` and `conditionAliases` so we don't have to do a workaround like this?

@RidhamShah @ppratikcr7 @VivekFitkariwala ?

Reply:
ok, maybe I've got it a little confused then what was showing up out of order exactly, it seems like there still should be a good way to save the order when submitting to backend instead of doing this work in UI. same for `conditionAliases` table in simple experiment.

Reply:
We will look into it.